### ISSUE TITLE
Configurable Products: Blank Image-Area after deselecting swatches/attributes on product-page

### ISSUE THREAD

**Issue by feelcomfort** [magento/magento2#18314]
### Preconditions

1. Update from Magento 2.2.4 to 2.2.6

### Steps to reproduce

1. Add Product in the Backend
2. Create Configurations
4. Use Attributes with the input type "Visual Swatch"
3. Save Product
4. Open Product-page in the frontend
5. Select Attribute/Swatch --> Images-Gallery of the simple product appears
6. Deselect Attribute/Swatch

### Expected result

1. Image-Gallery of the configurable product appears again
![expected_result](https://user-images.githubusercontent.com/43548743/46245234-d3531080-c3ea-11e8-9c3b-99e5834fad55.JPG)

### Actual result

1. Product page with a blank image-area
![actual_result](https://user-images.githubusercontent.com/43548743/46245237-de0da580-c3ea-11e8-90db-bd885f7e7f1c.JPG)

Does anyone have a solution/fix to this issue?
The only similar problem we found in web was a request without a solution, but 6 Month ago. Maybe it is not only a Magento 2.2.6 Bug: https://magento.stackexchange.com/questions/215228/when-i-deselect-the-swatch-once-after-selected-the-product-image-is-not-display

Thanks for help.

**Comment by magento-engcom-team** [comment]
Hi @feelcomfort. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@feelcomfort do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by okorshenko** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @okorshenko. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @okorshenko, here is your [Magento instance](https://i-18314-2-2-6.engcom.dev.magento.com).
Admin access: [https://i-18314-2-2-6.engcom.dev.magento.com/admin](https://i-18314-2-2-6.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by feelcomfort** [comment]
After testing different options: The issue depends on the attribute settings.

Using the catalog input type for store owner **"Dropdown": it works**
Saving the attribute as **"Visual Swatch": Blank image gallery**.

Attribute Properties:
![grafik](https://user-images.githubusercontent.com/43548743/46256506-175a1a00-c4ac-11e8-8fa1-40d21347af85.png)

Using Visual-Swatches:
![grafik](https://user-images.githubusercontent.com/43548743/46256720-d1528580-c4ae-11e8-9d7b-00f7106bf585.png)

Using Dropdown-Swatches:
![grafik](https://user-images.githubusercontent.com/43548743/46256709-b41db700-c4ae-11e8-8dd0-0c6bab53791c.png)

@magento-engcom-team: Can I ask for a 2.2.6 instance for testing too?

Another issue: In the past the swatch-images were saved and reloaded after changing to dropdown and back to visual-swatch. Since Mangento 2.2.6 the values are deleted and we need to reselect them. In the filespace this leads to copies of the same image in the same folder.

**Comment by raffoitaly** [comment]
Anyone can help us? Same problem for me in M 2.2.4

**Comment by neodrg** [comment]
Same error on our side on 2.2.6...

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-andrii. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @feelcomfort  I'm not able to reproduce following steps you described, seems like problem related to you custom extensions or code, if you like to update this please reopen the issue. Maybe it depends on your custom theme
